apps/web-giddh/src/app/contact/account-statement/account-statement.component.ts

```diff
@@ -0,0 +1,437 @@
+import { Component, OnDestroy, OnInit, ViewChild, Input, SimpleChanges, TemplateRef } from "@angular/core";
+import { MatDialog } from "@angular/material/dialog";
+import { MatSort } from "@angular/material/sort";
+import { MatPaginator, PageEvent } from '@angular/material/paginator';
+import { Observable, ReplaySubject } from "rxjs";
+import { debounceTime, distinctUntilChanged, takeUntil } from "rxjs/operators";
+import * as dayjs from 'dayjs';
+import { ContactComponentStore } from "../utility/contact.store";
+import { GIDDH_DATE_FORMAT, GIDDH_DATE_FORMAT_MM_DD_YYYY, GIDDH_NEW_DATE_FORMAT_UI } from "../../shared/helpers/defaultDateFormat";
+import { GIDDH_DATE_RANGE_PICKER_RANGES, PAGE_SIZE_OPTIONS } from "../../app.constant";
+import { BsModalRef, BsModalService } from 'ngx-bootstrap/modal';
+import { GeneralService } from "../../services/general.service";
+import { FormControl } from "@angular/forms";
+import { AdvanceSearchRequest } from "../../models/interfaces/advance-search-request";
+import { cloneDeep } from "../../lodash-optimized";
+
+
+@Component({
+    selector: "account-statement",
+    templateUrl: "account-statement.component.html",
+    styleUrls: ["account-statement.component.scss"],
+    providers: [ContactComponentStore]
+})
+export class AccountStatementComponent implements OnInit, OnDestroy {
+    /** Template reference for the datepicker modal */
+    @ViewChild('datepickerTemplate') public datepickerTemplate: TemplateRef<any>;
+    /** Template reference for the advance search modal */
+    @ViewChild('advanceSearchModal', { static: false }) public advanceSearchModal: any;
+    /** Reference to the Material paginator component */
+    @ViewChild(MatPaginator) paginator!: MatPaginator;
+    /** Reference to the Material sort component */
+    @ViewChild(MatSort) sort!: MatSort;
+    /** Holds localized JSON data specific to this module */
+    public localeData: any = {};
+    /** Holds common localized JSON data shared across modules */
+    public commonLocaleData: any = {};
+    /** True if API call is in progress */
+    public isLoading: boolean = false;
+    /** Used to unsubscribe all store listeners to avoid memory leaks */
+    private destroyed$: ReplaySubject<boolean> = new ReplaySubject(1);
+    /** Array of column names displayed in the account statement table */
+    public displayedColumns: string[] = ['Date', 'Transactions', 'Details', 'Amount', 'Payments', 'Balance'];
```

Comment:
```suggestion
    public displayedColumns: string[] = ['Date', 'Transactions', 'Details', 'Amount', 'Payments', 'Balance'];
```

use small case for all